.github/workflows/master.yml

```diff
@@ -15,86 +15,42 @@ permissions:
    contents: read
 
 jobs:
-   api_check:
-      runs-on: ubuntu-latest
-      if: github.repository == 'kotest/kotest'
-      steps:
-         -  name: Checkout the repo
-            uses: actions/checkout@v4
-            with:
-               fetch-depth: 0
-               ref: ${{ github.event.inputs.branch }}
-
-         -  name: Setup JDK
-            uses: actions/setup-java@v4
-            with:
-               distribution: 'temurin'
-               java-version: '17'
 
-         -  uses: gradle/actions/setup-gradle@v3
+   validate-abi:
+      name: Run ABI validations
+      if: github.repository == 'kotest/kotest'
+      uses: ./.github/workflows/run-gradle.yml
+      with:
+         runs-on: ubuntu-latest
+         ref: ${{ inputs.ref }}
+         task: apiCheck --continue
 
-         -  name: Run api check
-            run: ./gradlew apiCheck
 
-   linux:
-      needs: api_check
-      runs-on: ubuntu-latest
+   validate-all:
+      name: Run all Gradle validations
       if: github.repository == 'kotest/kotest'
+      needs: validate-abi
       strategy:
-         fail-fast: false
          matrix:
-            target:
-               - jvmTest publishJvmPublicationToDeployRepository publishKotlinMultiplatformPublicationToDeployRepository publishKotestBomPublicationToDeployRepository
-               - jsTest publishJsPublicationToDeployRepository
-               - wasmJsTest publishWasmJsPublicationToDeployRepository
-               - linuxX64Test publishLinuxX64PublicationToDeployRepository
-               - linuxArm64TestKlibrary publishLinuxArm64PublicationToDeployRepository
-               - androidNativeX86TestKlibrary androidNativeX64TestKlibrary publishAndroidNativeX64PublicationToDeployRepository publishAndroidNativeX86PublicationToDeployRepository
-               - androidNativeArm32TestKlibrary androidNativeArm64TestKlibrary publishAndroidNativeArm32PublicationToDeployRepository publishAndroidNativeArm64PublicationToDeployRepository
-      steps:
-         -  uses: kotest/kotest-action@master
-            with:
-               gradle-command: ${{ matrix.target }}
-               java-version: "17"
-
-   mac:
-      needs: api_check
-      runs-on: macos-latest
-      if: github.repository == 'kotest/kotest'
-      strategy:
+            include:
+               -  os: macos-latest
+               -  os: ubuntu-latest
+               -  os: windows-latest
```

Comment:
I think this will make us run jvmTest on all platforms. Currently we have one or two flaky tests that sometimes fails builds.. Will become important to fix that

Reply:
It'd be better to fix the flaky tests, or maybe enable retries for them?

We could also add `--exclude-task jvmTest` to 2 of the OSes, so Gradle will skip them, but of course fixing the flakey tests would be better!

Reply:
I'm ok with merging as-is. Let's try to reduce flakiness of master if it becomes an issue